src/components/SingleArticle.jsx

```diff
@@ -15,6 +17,36 @@ const SingleArticle = (props) => {
         article_img_url,
     } = props.article
 
+    const [userVotes, setUserVotes] = useState(0)
+    const [isLiked, setIsLiked] = useState(false)
+
+    const handleLike = (event) => {
+        event.preventDefault()
+        setUserVotes((currVotes) => {
+            if (!isLiked) {
+                return currVotes + 1
+            } else {
+                return currVotes - 1
+            }
+            
+        })
+        if (!isLiked) {
+            setIsLiked(true)
+            patchArticleVotes(article_id, 1)
+            .catch(() => {
+                setUserVotes(0)
+                setIsLiked(false)
+            })
+        } else {
+            setIsLiked(false)
+            patchArticleVotes(article_id, -1)
+            .catch(() => {
+                setUserVotes(1)
+                setIsLiked(true)
+            })
+        }
```

Comment:
I think ideally your catch block should be at the very end (after the else, not within the if and within the else blocks). Have a go at refactoring to make that drier :)